src/connection.js

```diff
@@ -522,7 +534,15 @@ class Connection extends EventEmitter {
     this.socket.on('close', this.socketClose);
     this.socket.on('end', this.socketEnd);
     this.messageIo = new MessageIO(this.socket, this.config.options.packetSize, this.debug);
-    this.messageIo.on('data', (data) => { this.dispatchEvent('data', data); });
+    this.messageIo.on('data', (data) => {
+      this.clearRequestTimer();
```

Comment:
Can you do a separate PR for this bug fix? Though this got caught as part of this work, I think we can call it unrelated.